skops/io/_numpy.py

```diff
@@ -24,13 +23,14 @@ def ndarray_get_state(obj, dst):
             res["type"] = "numpy"
             res["file"] = f_name
     except ValueError:
-        # object arrays cannot be saved with allow_pickle=False, therefore we
-        # convert them to a list and store them as a json file.
-        f_name = f"{uuid4()}.json"
-        with open(Path(dst) / f_name, "w") as f:
-            f.write(json.dumps(obj.tolist()))
+        # Object arrays cannot be saved with allow_pickle=False, therefore we
+        # convert them to a list and recursively call get_state on it.
+        if obj.dtype == object:
+            obj = get_state(obj.tolist(), dst)
+            res["content"] = obj["content"]
```

Comment:
Oh wait, how large are these, you sure we should put them in the schema.json?

Reply:
Since we call `get_state`, if there are any numpy arrays contained, they'll be put on disk, so we should be good, right?